### PR TITLE
chore: Remove lifecycle warnings and error on lintr checks

### PR DESCRIPTION
**Pull request by schloerke**
## Checklist

- [NA] Does this change update `NEWS.md` (referencing the connected issue if necessary)?
- [NA] Does this change need documentation? Have you run `document()?

### CODE REVIEW

**Comment by schloerke** [comment]
@toph-allen Assigning PR to you to finish out as you see fit. Thank you!

**Comment by toph-allen** [comment]
@schloerke I increased the line length linter to 120 to cut down on the number of required fixes and added a few other exceptions, and then just powered through all the errors. Hopefully nothing broke!

File: R/parse.R [hunk 1]
```diff
@@ -168,12 +170,12 @@ parse_connect_rfc3339 <- function(x) {
   })
 }
 
-vec_cast.POSIXct.double <- function(x, to, ...) {
+vec_cast.POSIXct.double <- function(x, to, ...) { # nolint: object_name_linter
   warn_experimental("vec_cast.POSIXct.double")
```
Comment (nealrichardson) [inline R/parse.R:174]: I was confused by the PR title: I thought it meant you were going to remove all of these chatty/panicky warnings.

Comment (schloerke) [reply R/parse.R:174]: If I remove the lint, this error is displayed: 
```
Variable and function name style should match snake_case, symbols or CamelCase. lintr(object_name_linter)
```

I believe this is a false-positive find as the name before the `.` is valid and the lint should be suppressed.

Comment (schloerke) [reply R/parse.R:174]: With this PR's changes, no warnings are being displayed within the PR files after GitHub Actions runs. 

(Resolving)

**Comment by schloerke** [comment]
Thank you @toph-allen !

**Review by toph-allen — APPROVED** [review]

File: R/parse.R (hunk 1 above)
Comment (schloerke) [reply R/parse.R]: Ah! Makes sense. Thank you!

Yes. ... Different code quality warnings that should be surfaced / fixed.

**Pull request merged**